src/modules/claims/claims.service.ts

```diff
@@ -521,16 +521,22 @@ export class ClaimsService {
     this.validatePublishPublicClaimRequest(registrationTypes, claim);
     let url: string | undefined = undefined;
     if (registrationTypes.includes(RegistrationTypes.OnChain)) {
+      if (!claim.claimType) {
+        throw new Error(ERROR_MESSAGES.CLAIM_TYPE_MISSING);
+      }
+
       const claims = await this.getClaimsBySubject({
         did: this._signerService.did,
-        namespace: claim.claimType,
+        // get namespace from claimType
+        namespace: claim.claimType.split('.').slice(2).join('.'),
```

Comment:
isn't it better to create method or function `getNamespaceFromClaimType(claimType: string): string` ? This way there is no need to create a comment.